### PR TITLE
Check keys if revoke certificate by private key 

### PR DESCRIPTION
**Pull request by Baeumla**
Issue #4805

Revoke certificate by key now checks if the provided private key matches the certificate.

### CODE REVIEW

**Comment by Baeumla** [comment]
@ynasser 

Since the original function threw the error `Details: [('SSL routines', 'SSL_CTX_check_private_key', 'no private key assigned')]` which was weird to begin with, I re-wrote the function `verify_cert_matches_priv_key` to call the OpenSSL function directly with the key and cert file path (default filetype PEM). 

Now, Travis throws this error, and I really don't know how that could happen or what I should do about it. Is it possible that the certificate really does not match the provided private key?
`Error: verifying the cert located at /home/travis/build/certbot/certbot/certbot/tests/testdata/cert.pem matches the                 private key located at /home/travis/build/certbot/certbot/certbot/tests/testdata/rsa256_key.pem has failed.                 Details: [('x509 certificate routines', 'X509_check_private_key', 'key values mismatch')]`

**Comment by ynasser** [comment]
@Baeumla

Great job so far!

The problems you're experiencing stem from the fact that we have a horribly organized `certbot/tests/testdata/` directory, and it's not clear which of our testing certificates go with which keys. The reason why `test_revoke_with_key` in `certbot/tests/main_test.py` is failing is because the new verification code you wrote is in fact working properly. I don't think `cert.pem` was generated with `rsa256_key.pem`, as you've pointed out. I think that *previously* any values would have worked in that test, as the only things that mattered were reasonable-looking values existing. I do think we need some more thorough tests for revoking via a key, which is part of the issue you are working on :)

In tests which will trigger the changes you're making, I'd just copy `self_signed_cert.pem` and `rsa2048_key.pem` from `certbot/tests/crypto_util_test.py`, as for those only values which actually match can be used.

By the way, something useful to know which _may_ come up is knowing the differences between `cert`, `chain`, and `fullchain`, which is explained here: https://certbot.eff.org/docs/using.html#where-are-my-certificates.

**Comment by ynasser** [comment]
FYI: I've made an issue for cleaning up + documenting `certbot/tests/testdata/` in: #4917.

File: certbot/tests/main_test.py [hunk 1]
```diff
@@ -1012,12 +1014,12 @@ def test_certonly_csr_dry_run(self):
     @mock.patch('certbot.main.client.acme_client')
     def test_revoke_with_key(self, mock_acme_client):
```
Comment (ynasser) [inline certbot/tests/main_test.py]: I'd add another test below this one named something like: `test_revoke_with_mismatched_key` and make sure the appropriate error is raised to make sure revoking by key is failing when it's supposed to. You can use the previous values of `CERT` and `KEY` to make it fail.

Comment (Baeumla) [inline certbot/tests/main_test.py]: Yeah that makes sense. Should have thought about that before already. Sorry that it took me so long for the commit but I was having a stressful week at work.

**Comment by ynasser** [comment]
This looks good to me!

@jsha since you originally requested this feature, do you want to take a quick look to make sure this is what you had in mind when you requested this feature?

**Comment by ynasser** [comment]
Thanks for this @Baeumla!

**Pull request merged**